using-directus-as-a-baby-health-tracker/index.md

```diff
@@ -0,0 +1,362 @@
+#### As a Widget for iOS
+As a last method of reporting and integration Directus, we can create a simple iOS widget, that sits right on my lock and home screen. As this is not about iOS development we chose the easy way and used Scriptable for this. This free app can be used to run JavaScript and display the data as a widget. The only downside of these Scriptable widgets is, that iOS decides WHEN to update the content of the widget. So you can not force the widget to reload actively. This is a limitation to prevent battery drainage and high load due to the demanding JavaScript. Usually, the script runs every 1 to 15 minutes. To know how "outdated" the displayed data is, we've added the create-timestamp as part of the widgets. 
+For the Directus integration this time I have created a new role that can have view access to the data collection only. A user in this role has set a static token that I can use within the widget. Using the query parameter `limit` and `sort` I can define to get only the very last entry sorted by the create date column.
+
+```javascript
+const url = "https://my-directus.app/items/neebo_data?limit=1&sort=sort,-date_created"
+const token = "xy-my-static-token"
+async function getData(url) {
+  let request = new Request(url);
+  let authHeader = "Bearer " + token
+  request.headers = { "Authorization": authHeader }
+
+  let data = await request.loadJSON();
+  return data;
+}
+function processData(data) {
+  console.log(data)
+  let processedData = {
+    "placement": data.data[0].placement,
+    "battery": {
+      "level": data.data[0].battery_level,
+      "state": data.data[0].battery_state
+    },
+    "termal": {
+      "state": "",
+      "value": data.data[0].temperature
+    },
+    "oxygen": {
+      "state": "",
+      "value": data.data[0].oxygen
+    },
+    "heart_rate": {
+      "state": "",
+      "value": data.data[0].heart_rate
+    },
+    "reldate": new RelativeDateTimeFormatter().string(new Date(data.data[0].date_created), new Date()),
+    "datecreated":
+      new Date(data.data[0].date_created).getHours() + ":" +
+      String(new Date(data.data[0].date_created).getMinutes()).padStart(2, "0")
+  }
+
+  return processedData;
+}
+function createWidget(data) {
+  var widget = new ListWidget();
+  console.log(data.placement);
+  switch (data.placement) {
+    case 4:  // Charging
+      var text = widget.addText("⚡️ " + data.battery.level + "% …");
+      break;
+    case "2":
+      break;
+    default:
+      var text = widget.addText("🌡️ " + data.termal.value + "°C");
+      var text = widget.addText("🫧 " + data.oxygen.value + " ♥️ " + data.heart_rate.value);
+      break;
+  }
+  var text = widget.addText(data.datecreated);
+  text.font = Font.regularRoundedSystemFont(12);
+  return widget;
+}
+let data = await getData(url);
+let processedData = processData(data);
+let widget = await createWidget(processedData);
+Script.setWidget(widget);
+Script.complete()
+```
```

Comment:
````suggestion
## Build an iOS Widget
As a last method of reporting and integration Directus, we can create an iOS widget, that sits right on my lock and home screen. As this is not about iOS development we chose the easy way and used Scriptable for this. This free app can be used to run JavaScript and display the data as a widget. The only downside of these Scriptable widgets is, that iOS decides WHEN to update the content of the widget. So you can not force the widget to reload actively. This is a limitation to prevent battery drainage and high load due to the demanding JavaScript. Usually, the script runs every 1 to 15 minutes. To know how "outdated" the displayed data is, we've added the create-timestamp as part of the widgets. 

For the Directus integration this time I have created a new role that can have view access to the data collection only. A user in this role has set a static token that I can use within the widget. Using the query parameter `limit` and `sort` I can define to get only the very last entry sorted by the create date column.

```javascript
const url = "https://my-directus.app/items/sensor_data?limit=1&sort=sort,-date_created"
const token = "xy-my-static-token"
async function getData(url) {
  let request = new Request(url);
  let authHeader = "Bearer " + token
  request.headers = { "Authorization": authHeader }

  let data = await request.loadJSON();
  return data;
}
function processData(data) {
  console.log(data)
  let processedData = {
    "placement": data.data[0].placement,
    "battery": {
      "level": data.data[0].battery_level,
      "state": data.data[0].battery_state
    },
    "termal": {
      "state": "",
      "value": data.data[0].temperature
    },
    "oxygen": {
      "state": "",
      "value": data.data[0].oxygen
    },
    "heart_rate": {
      "state": "",
      "value": data.data[0].heart_rate
    },
    "reldate": new RelativeDateTimeFormatter().string(new Date(data.data[0].date_created), new Date()),
    "datecreated":
      new Date(data.data[0].date_created).getHours() + ":" +
      String(new Date(data.data[0].date_created).getMinutes()).padStart(2, "0")
  }

  return processedData;
}
function createWidget(data) {
  var widget = new ListWidget();
  console.log(data.placement);
  switch (data.placement) {
    case 4:  // Charging
      var text = widget.addText("⚡️ " + data.battery.level + "% …");
      break;
    case "2":
      break;
    default:
      var text = widget.addText("🌡️ " + data.termal.value + "°C");
      var text = widget.addText("🫧 " + data.oxygen.value + " ♥️ " + data.heart_rate.value);
      break;
  }
  var text = widget.addText(data.datecreated);
  text.font = Font.regularRoundedSystemFont(12);
  return widget;
}
let data = await getData(url);
let processedData = processData(data);
let widget = await createWidget(processedData);
Script.setWidget(widget);
Script.complete()
```
````

Reply:
Could you add a quick screenshot of the widget here at the end?